File: CloudFormation/src/main/java/com/intellij/aws/cloudformation/CloudFormationReferenceContributor.java
package com.intellij.aws.cloudformation;

import com.intellij.json.psi.JsonStringLiteral;
import com.intellij.patterns.PlatformPatterns;
import com.intellij.psi.PsiReferenceContributor;
import com.intellij.psi.PsiReferenceRegistrar;
import org.jetbrains.annotations.NotNull;

public class CloudFormationReferenceContributor extends PsiReferenceContributor {
  private CloudFormationReferenceProvider ReferenceProviderInstance = new CloudFormationReferenceProvider();

  @Override
  public void registerReferenceProviders(@NotNull PsiReferenceRegistrar registrar) {
    registrar.registerReferenceProvider(PlatformPatterns.psiElement(JsonStringLiteral.class), ReferenceProviderInstance);
  }
}

File: CloudFormation/src/main/kotlin/com/intellij/aws/cloudformation/CloudFormationReferenceContributor.kt
package com.intellij.aws.cloudformation

import com.intellij.json.psi.JsonReferenceExpression
import com.intellij.json.psi.JsonStringLiteral
import com.intellij.patterns.PlatformPatterns
import com.intellij.psi.PsiReferenceContributor
import com.intellij.psi.PsiReferenceRegistrar
import org.jetbrains.yaml.psi.YAMLScalar

class CloudFormationReferenceContributor : PsiReferenceContributor() {
  private val ReferenceProviderInstance = CloudFormationReferenceProvider()

  override fun registerReferenceProviders(registrar: PsiReferenceRegistrar) {
    registrar.registerReferenceProvider(PlatformPatterns.psiElement(JsonStringLiteral::class.java), ReferenceProviderInstance)
    registrar.registerReferenceProvider(PlatformPatterns.psiElement(JsonReferenceExpression::class.java), ReferenceProviderInstance)
    registrar.registerReferenceProvider(PlatformPatterns.psiElement(YAMLScalar::class.java), ReferenceProviderInstance)
  }
}
